tools/src/commands/EasDispatch.ts

```diff
@@ -436,66 +396,292 @@ async function internalRemoveBackgroundPermissionsFromInfoPlistAsync(): Promise<
   await fs.writeFile(INFO_PLIST_PATH, plist.build(parsedPlist));
 }
 
-async function internalIosSimulatorPublishAsync() {
-  const tmpTarGzPath = path.join(os.tmpdir(), 'simulator.tar.gz');
+async function androidApkBuildAsync() {
   const projectDir = path.join(EXPO_DIR, 'apps/eas-expo-go');
+  await enforceRunningOnSdkReleaseBranchAsync();
+  await prepareAndroidCredentialsAsync(projectDir);
+
+  await spawnAsync(
+    'eas',
+    ['build', '--platform', 'android', '--profile', PUBLISH_CLIENT_BUILD_PROFILE],
+    {
+      cwd: projectDir,
+      stdio: 'inherit',
+      env: {
+        ...process.env,
+        EAS_DANGEROUS_OVERRIDE_ANDROID_APPLICATION_ID: 'host.exp.exponent',
+      },
+    }
+  );
+}
+
+async function androidApkUploadAsync() {
+  if (!process.env.GITHUB_TOKEN) {
+    logger.error('GITHUB_TOKEN is not set. Please set it in your environment.');
+    return;
+  }
+
   const sdkVersion = await enforceRunningOnSdkReleaseBranchAsync();
-  const artifactPaths = glob.sync('ios/build/Build/Products/*simulator/*.app', {
-    absolute: true,
-    cwd: projectDir,
-  });
+  const appVersion = await androidAppVersionAsync();
+  await confirmPromptIfOverridingRemoteFileAsync(getAndroidApkUrl(appVersion), appVersion);
+  const projectDir = path.join(EXPO_DIR, 'apps/eas-expo-go');
```

Comment:
not related but we can move `path.join(EXPO_DIR, 'apps/eas-expo-go');` as a constant at top